stores/migrations.go

```diff
@@ -178,7 +185,15 @@ func performMigrations(db *gorm.DB, logger glogger.Interface) error {
 // initSchema is executed only on a clean database. Otherwise the individual
 // migrations are executed.
 func initSchema(tx *gorm.DB) error {
-	return tx.AutoMigrate(tables...)
+	err := tx.AutoMigrate(tables...)
+	if err != nil {
+		return err
+	}
+	// Change the object_id colum to use case sensitive collation.
+	if !isSQLite(tx) {
+		return tx.Exec("ALTER TABLE objects MODIFY COLUMN object_id VARCHAR(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_bin;").Error
```

Comment:
I'm wondering whether we want to set char set and collate settings as our default table options. Wouldn't that be better? Gorm allows doing that by doing something like: 

`db = db.Set("gorm:table_options", "ENGINE=InnoDB CHARSET=utf8 COLLATE=utf8_bin")`

Reply:
I think there is also a way to do that using tags on the field but I haven't been able to figure out how to do it in a cross-db way that works for both SQLite and MySQL. Usually one of them breaks.

Reply:
Hm yeah I think so too but couldn't immediately find it. Regardless though shouldn't we pass the table options rather than executing an `ALTER` statement. Especially since we're initializing the schema here.

We can do  `return tx.Set("gorm:table_options", "CHARSET=utf8mb4 COLLATE=utf8mb4_bin").AutoMigrate()` so it'll apply on all of our tables, bound to run into this again at one point?